R/NielsonKuparuk.R

```diff
@@ -0,0 +1,278 @@
+}
+
+#Format Date_adt
+for(i in 1:length(files)){
+  format(as.POSIXct(files[[i]]$Date_adt, tz = ""), format = "%Y-%m-%d %H:%M:%S")
+}
+
+#Data file editing complete!
+
+#Reassign edited files to new file names
+write.csv(files[[1]], file = "/home/mnguyen/Neilson/Neilson_IC_WeirQ.csv", row.names = FALSE) 
+write.csv(files[[2]], file = "/home/mnguyen/Neilson/Neilson_ICKUP_Q.csv", row.names = FALSE) 
+write.csv(files[[3]], file = "/home/mnguyen/Neilson/Neilson_Kup2AQ.csv", row.names = FALSE) 
+write.csv(files[[4]], file = "/home/mnguyen/Neilson/Neilson_Kup2TQ.csv", row.names = FALSE) 
+write.csv(files[[5]], file = "/home/mnguyen/Neilson/Neilson_Kup3BQ.csv", row.names = FALSE) 
+write.csv(files[[6]], file = "/home/mnguyen/Neilson/Neilson_Kup3TQ.csv", row.names = FALSE) 
+write.csv(files[[7]], file = "/home/mnguyen/Neilson/Neilson_Kup4Q.csv", row.names = FALSE) 
+write.csv(files[[8]], file = "/home/mnguyen/Neilson/Neilson_Kup5Q.csv", row.names = FALSE) 
+write.csv(files[[9]], file = "/home/mnguyen/Neilson/Neilson_Kup8Q.csv", row.names = FALSE) 
+write.csv(files[[10]], file = "/home/mnguyen/Neilson/Neilson_Kup9Q.csv", row.names = FALSE) 
+
+
+#Check if length and values the same
+test <- read.csv("/home/mnguyen/Neilson/Neilson_Kup9Q.csv", header=TRUE)
+test$Discharge_cms == Kup9Q$Discharge_cms
+length(test$Discharge_cms) == length(Kup9Q$Discharge_cms)
+#Good to go
+
+#EML
+
+eml <- read_eml("/home/mnguyen/Neilson/NeilsonMetadata.xml")
+
+#create attributes table
+
+library(RCurl)
+script <- getURL("https://raw.githubusercontent.com/NCEAS/datamgmt/master/R/create_attributes_table.R", ssl.verifypeer = FALSE)
+eval(parse(text = script))
+
+library(rhandsontable)
+library(shiny)
+
+df <- read.csv("/home/mnguyen/Neilson/Neilson_ICKUP_Q.csv")
+create_attributes_table(df)
+
+attributes <- data.frame(
+  attributeName = c('Date_adt','Discharge_cms'),
+  domain = c('dateTimeDomain','numericDomain'),
+  attributeDefinition = c('Date and time of sample','Sample discharge'),
+  definition = c(NA,NA),
+  measurementScale = c('dateTime','ratio'),
+  formatString = c('YYYY-MM-DD hh:mm:ss',NA),
+  numberType = c(NA,'real'),
+  unit = c(NA,'centimeter'),
+  missingValueCode = c(NA,NA),
+  missingValueCodeExplanation = c(NA,NA),
+  stringsAsFactors = FALSE)
+
+attributeList1 <- set_attributes(attributes)
+
+#Create new datapids
+
+data_pathIC_WeirQ <- "/home/mnguyen/Neilson/Neilson_IC_WeirQ.csv"
+data_pathICKUP_Q <- "/home/mnguyen/Neilson/Neilson_ICKUP_Q.csv"
+data_pathKup2AQ <- "/home/mnguyen/Neilson/Neilson_Kup2AQ.csv"
+data_pathKup2TQ <- "/home/mnguyen/Neilson/Neilson_Kup2TQ.csv"
+data_pathKup3BQ <- "/home/mnguyen/Neilson/Neilson_Kup3BQ.csv"
+data_pathKup3TQ <- "/home/mnguyen/Neilson/Neilson_Kup3TQ.csv"
+data_pathKup4Q <- "/home/mnguyen/Neilson/Neilson_Kup4Q.csv"
+data_pathKup5Q <- "/home/mnguyen/Neilson/Neilson_Kup5Q.csv"
+data_pathKup8Q <- "/home/mnguyen/Neilson/Neilson_Kup8Q.csv"
+data_pathKup9Q <- "/home/mnguyen/Neilson/Neilson_Kup9Q.csv"
```

Comment:
If you're copying and pasting stuff a bunch of times, it's a sign that you can write a function/vectorize. I also like to write things out sometimes just to make things go where they should go, but if you feel like you're ready to "level-up" into function writing, it can make life easier in the long run. No pressure to rush into it, but it's just something to keep in mind and test out.

An example function for your first two steps might be:
```
your_function <- function(data_path) {
    pid <- publish_object(mn, data_path, format_id = "text/csv")
    pid_to_eml_physical(mn, pid)
}
```

Then you can do something like:
```
data_paths <- list.files(path = DIRECTORY_PATH, pattern = "csv", full.names = TRUE) 
#grab all files in your directory that have "csv" in the file name; get the full file extension (full.names = TRUE)

physicals <- lapply(data_paths, your_function)
```